client.go

```diff
@@ -77,21 +117,27 @@ func (c *StatsdClient) String() string {
 	return c.addr
 }
 
+// Reconnect causes the client to re-create its underlying socket used to send
+// stats. Any existing socket is closed before opening a new socket. Any error
+// encountered while closing an existing socket is logged, but not returned.
 func (c *StatsdClient) Reconnect() error {
-	var err error
-	if c.conn_type == "udp" {
-		fmt.Println("creating new udp")
-		err = c.CreateSocket()
-	} else if c.conn_type == "tcp" {
-		fmt.Println("creating new tcp")
-		err = c.CreateTCPSocket()
-	} else if c.conn_type == "" {
-		return fmt.Errorf("No socket created, cannot identify connection type")
+	// close any existing socket before creating a new one
+	if c.conn != nil {
+		if err := c.conn.Close(); err != nil {
```

Comment:
I was thinking "should we close the connection after we've successfully recreated the new one?   like.. in line 160?   Then if somehow `net.DialTimeout` returns err, we could still use the existing one.

Reply:
@jsongHBO Added that.  PTAL.